### PR TITLE
Allow watching for ready status

### PR DESCRIPTION
**Pull request by joesis**
For https://github.com/getlantern/lantern-internal/issues/2965

It mainly moved some feature from the direct instance to the fronting context, so we can watch for the ready status directly without worrying about reconfiguration.

Also initialized go module.

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -39,15 +46,19 @@ func CloseCache() {
...
 
 func NewFrontingContext(name string) *FrontingContext {
-	return &FrontingContext{
-		name:     name,
-		instance: eventual.NewValue(),
+	fc := &FrontingContext{
+		name:  name,
+		ready: make(chan struct{}),
 	}
+	fc._instance.Store(newDirect(nil, 0, "", tls.ClientHelloID{}, fc.signalReady))
```
Comment (joesis) [inline context.go]: Always have a (null) direct instance.

File: direct.go [hunk 2]
```diff
@@ -87,25 +95,6 @@ func (d *direct) providerFor(m *masquerade) *Provider {
 	return d.providers[pid]
 }
 
-// Vet vets the specified Masquerade, verifying certificate using the given CertPool
```
Comment (joesis) [inline direct.go]: Moved to the bottom of the file. It doesn't seem to be used anywhere except in the tests. Do you by chance remember why it was introduced @oxtoacart ?

Comment (oxtoacart) [reply direct.go]: Sorry, I don't remember. I don't see it being used anywhere either (I checked flashlight and checkfallbacks). I think it's okay to remove if you want.

Comment (joesis) [reply direct.go]: Thanks. Let me just leave it at the bottom.

**Comment by forkner** [comment]
@joesis for some reason `TestCustomValidators` seems to be failing on this branch for me (and not master) ... possibly this just changes some expectation of the test

**Comment by forkner** [comment]
@joesis ah -- I think that test expects it can call Configure(...) more than once -- which may not quite work as expected with the ready signal on the context? -- I believe Configure() can be called multiple times in a run of the application (ie when a new configuration is received)

Possibly we should add an explicit test here for that to makes sure it's doing what we'd expect.

**Comment by forkner** [comment]
Also just generally, it looks like we should add this to CI and possibly refresh some of the test data which is super stale and makes the tests run long.  I'll add a separate ticket for that.

**Comment by joesis** [comment]
> @joesis ah -- I think that `TestCustomValidators` expects it can call Configure(...) more than once -- which may not quite work as expected with the ready signal on the context? -- I believe Configure() can be called multiple times in a run of the application (ie when a new configuration is received)
> 
> Possibly we should add an explicit test here for that to makes sure it's doing what we'd expect.

Oh I thought when reconfiguring, the cache should contain some valid masquerades before the new ones getting vetted, but it's not always true especially in the test, which doesn't have cache file. Good catch!

**Pull request closed without merging**